public/app/features/templating/template_srv.ts

```diff
@@ -270,6 +278,13 @@ export class TemplateSrv implements BaseTemplateSrv {
         return match;
       }
 
+      if (isAdHoc(variable)) {
+        const value = variable.filters;
+        const text = variable.id;
+
+        return this.formatValue(value, fmt, variable, text);
+      }
```

Comment:
Feels like the canHandle function adds some unnecessary complexity/duplication to all formats for just this scenario (at least for now), so instead of having canHandle on all formats can we not just allow it? if we here set value to JSON.stringify(variable.filters) all the other formatters will handkle adhoc fine, and the queryParm will still work the same as it's not use value but inspecting the variable object

Reply:
Great suggestion, I'll fix!

Reply:
Fixed!